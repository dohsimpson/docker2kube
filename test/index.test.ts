import convert from '../src/index'
import { stringify } from 'yaml';

describe('convert', () => {
  it('should work', () => {
    const yaml = `
    services:
      frontend:
        labels:
          - "com.example.description=Accounting webapp"
          - "com.example.department=Finance"
          - "com.example.label-with-empty-value"
        image: awesome/webapp
        ports:
          - "443:8043"
        expose:
          - "3000"
        hostname: my-frontend
        networks:
          - front-tier
          - back-tier
        configs:
          - httpd-config
        secrets:
          - server-certificate
        command: echo hello world
        deploy:
          replicas: 3
          resources:
            limits:
              cpus: '0.001'
              memory: 50M
            reservations:
              cpus: '0.0001'
              memory: 20M
          restart_policy:
            condition: on-failure
            delay: 5s
            max_attempts: 3
            window: 120s
          placement:
            constraints:
              - node.role == manager
              - engine.labels.operatingsystem == ubuntu 14.04
        environment:
          RACK_ENV: development
          SHOW: "true"
          USER_INPUT:
        cap_add:
          - ALL
        tmpfs:
          - /tmp
          - /run
        stop_grace_period: 1m30s
        healthcheck:
          test: ["CMD", "curl", "-f", "http://localhost:8080"]
          interval: 30s
          timeout: 10s
          retries: 3

      backend:
        container_name: my-backend
        image: awesome/database
        volumes:
          - db-data:/etc/data
        networks:
          - back-tier
        environment:
          - RACK_ENV=development
          - SHOW=true
          - USER_INPUT
        configs:
          - source: httpd-config
            target: /etc/apache2/httpd.conf
            uid: '1000'
            gid: '1000'
            mode: 0440

    volumes:
      db-data:
        driver: flocker
        driver_opts:
          size: "10GiB"

    configs:
      httpd-config:
        file: ./httpd.conf

    secrets:
      server-certificate:
        external: true

    networks:
      # The presence of these objects is sufficient to define them
      front-tier: {}
      back-tier: {}
    `
    const result = convert(yaml);
    expect(result).toMatchSnapshot();
  });
})
