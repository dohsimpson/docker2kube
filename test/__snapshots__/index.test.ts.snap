// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert should work 1`] = `
"---
apiVersion: v1
kind: Service
metadata:
  namespace: default
  name: frontend-443
  labels:
    app: frontend
spec:
  ports:
    - appProtocol: TCP
      targetPort: 8043
      port: 443
  selector:
    app: frontend
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: frontend
  namespace: default
  labels:
    app: frontend
  annotations:
    com.example.description: Accounting webapp
    com.example.department: Finance
    com.example.label-with-empty-value: ""
spec:
  selector:
    matchLabels:
      app: frontend
  template:
    metadata:
      labels:
        app: frontend
    spec:
      containers:
        - name: frontend
          image: awesome/webapp
          command:
            - echo
            - hello
            - world
          volumeMounts:
            - name: httpd-config
              mountPath: /httpd-config
              readOnly: true
            - name: tmp
              mountPath: /tmp
            - name: run
              mountPath: /run
          ports:
            - containerPort: 8043
              protocol: TCP
            - containerPort: 3000
              protocol: TCP
          env:
            - name: RACK_ENV
              value: development
            - name: SHOW
              value: "true"
            - name: USER_INPUT
          securityContext:
            capabilities:
              add:
                - ALL
          resources:
            limits:
              cpu: 1m
              memory: 50M
            requests:
              cpu: 0.1m
              memory: 20M
          livenessProbe:
            periodSeconds: 30
            timeoutSeconds: 10
            failureThreshold: 3
            exec:
              command:
                - curl
                - -f
                - http://localhost:8080
      hostname: my-frontend
      terminationGracePeriodSeconds: 90
      volumes:
        - name: httpd-config
          configMap:
            name: httpd-config
        - name: tmp
          emptyDir: {}
        - name: run
          emptyDir: {}
  replicas: 3
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: my-backend
  namespace: default
  labels:
    app: my-backend
  annotations: {}
spec:
  selector:
    matchLabels:
      app: my-backend
  template:
    metadata:
      labels:
        app: my-backend
    spec:
      containers:
        - name: my-backend
          image: awesome/database
          volumeMounts:
            - name: db-data
              mountPath: /etc/data
            - name: httpd-config
              mountPath: /etc/apache2/httpd.conf
              readOnly: true
          ports: []
          env:
            - name: RACK_ENV
              value: development
            - name: SHOW
              value: "true"
            - name: USER_INPUT
      volumes:
        - name: db-data
          emptyDir: {}
        - name: httpd-config
          configMap:
            name: httpd-config
  replicas: 1
"
`;
